Link errors building KokkosSparse_gs.exe perf test

With the kokkos-kernels develop branch I'm unable to build the gauss-seidel performance test `KokkosSparse_gs.cpp` due to link errors:

<details>
 <summary>LinkErrors</summary>

```
g++  KokkosSparse_gs.o -L/Users/ndellin/Research/kokkos_items/kokkos-kernels/testing/ETILinkError/install/lib -L./ -lkokkos_kernels -lkokkos -    ldl -lpthread -L/Users/ndellin/Research/kokkos_items/kokkos-kernels/testing/ETILinkError/kokkos/install/lib -fopenmp   -o KokkosSparse_gs.exe
Undefined symbols for architecture x86_64:
  "KokkosSparse::Impl::GAUSS_SEIDEL_APPLY<KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long const, int const, double const, Kokkos:: OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::     HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::  MemoryTraits<1u> >, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u>   >, Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<double  const**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, false, false>::                       gauss_seidel_apply(KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long const, int const, double const, Kokkos::OpenMP, Kokkos::        HostSpace, Kokkos::HostSpace>*, int, int, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::          HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::  MemoryTraits<1u> >, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u>   >, Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<double  const**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, bool, bool, double, int, bool,        bool)", referenced from:
      void KokkosSparse::Experimental::symmetric_gauss_seidel_apply<KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long, int, double,  Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP,      Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<double*, Kokkos::LayoutRight, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, void>, Kokkos::View<double*,    Kokkos::LayoutRight, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<double*, Kokkos::LayoutRight,    Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> > >(KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long,   int, double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>*, KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long, int, double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>::const_nnz_lno_t, KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long, int,      double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>::const_nnz_lno_t, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::   Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP,      Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<double*, Kokkos::LayoutRight, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>,     void>, Kokkos::View<double*, Kokkos::LayoutRight, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::         View<double*, Kokkos::LayoutRight, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, bool, bool, KokkosKernels::     Experimental::KokkosKernelsHandle<unsigned long, int, double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>::nnz_scalar_t, int) in        KokkosSparse_gs.o
  "KokkosSparse::Impl::GAUSS_SEIDEL_NUMERIC<KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long const, int const, double const,        Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP,      Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>,   Kokkos::MemoryTraits<1u> >, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::           MemoryTraits<1u> >, false, false>::gauss_seidel_numeric(KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long const, int const, double   const, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>*, int, int, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::          Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::       OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::   HostSpace>, Kokkos::MemoryTraits<1u> >, bool)", referenced from:
      void KokkosSparse::Experimental::gauss_seidel_numeric<KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long, int, double, Kokkos:: OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::     HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::        MemoryTraits<0u> >, Kokkos::View<double*, Kokkos::LayoutRight, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, void> >(KokkosKernels::         Experimental::KokkosKernelsHandle<unsigned long, int, double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>*, KokkosKernels::             Experimental::KokkosKernelsHandle<unsigned long, int, double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>::const_nnz_lno_t,             KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long, int, double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>::              const_nnz_lno_t, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::               MemoryTraits<0u> >, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<double*, Kokkos::LayoutRight, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, void>, bool) in KokkosSparse_gs.o
  "KokkosSparse::Impl::GAUSS_SEIDEL_SYMBOLIC<KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long const, int const, double const,       Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP,      Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>,   Kokkos::MemoryTraits<1u> >, false, false>::gauss_seidel_symbolic(KokkosKernels::Experimental::KokkosKernelsHandle<unsigned long const, int const, double const, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>*, int, int, Kokkos::View<unsigned long const*, Kokkos::LayoutLeft, Kokkos::   Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::       OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, bool)", referenced from:
      void runGS<unsigned long, int, Kokkos::OpenMP>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::       __cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in KokkosSparse_gs.o
  "KokkosSparse::Impl::SPMV<double const, int const, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u>, unsigned long   const, double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<3u>, double*, Kokkos::          LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u>, false, false>::spmv(char const*, double const&,          KokkosSparse::CrsMatrix<double const, int const, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u>, unsigned long       const> const&, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<3u> >      const&, double const&, Kokkos::View<double*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >    const&)", referenced from:
      void runGS<unsigned long, int, Kokkos::OpenMP>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::       __cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in KokkosSparse_gs.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
```
</details>

develop branch SHA: 4cd9627e5bbe2bef8554c7bc6ea637bf849df2b8

OpenMP build with gcc/6.5 on my Mac laptop

Configuration:
```
KOKKOS_PATH=${HOME}/kokkos #path to kokkos source
KOKKOSKERNELS_SCALARS=double
KOKKOSKERNELS_LAYOUTS=LayoutLeft #the layout types to instantiate.
KOKKOSKERNELS_ORDINALS=int #ordinal types to instantiate
KOKKOSKERNELS_OFFSETS=int #offset types to instantiate
KOKKOSKERNELS_PATH=../.. #path to kokkos-kernels top directory.
KOKKOSKERNELS_OPTIONS=eti-only #options for kokkoskernels  
CXXFLAGS="-Wall -pedantic -Werror -O3 -g -Wshadow -Wsign-compare -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized"
CXX=g++ #${KOKKOS_PATH}/bin/nvcc_wrapper
KOKKOS_DEVICES=OpenMP #devices Cuda...

../../scripts/generate_makefile.bash --kokkoskernels-path=${KOKKOSKERNELS_PATH} --with-scalars=${KOKKOSKERNELS_SCALARS} --with-                   ordinals=${KOKKOSKERNELS_ORDINALS} --with-offsets=${KOKKOSKERNELS_OFFSETS} --kokkos-path=${KOKKOS_PATH} --with-devices=${KOKKOS_DEVICES} --       arch=${KOKKOS_ARCHS} --compiler=${CXX} --with-options=${KOKKOSKERNELS_OPTIONS}  --cxxflags="${CXXFLAGS}" --with-kokkos-                           options=${KOKKOS_OPTIONS} --with-layouts=${KOKKOSKERNELS_LAYOUTS}
```

--- a/perf_test/sparse/KokkosSparse_gs.cpp
+++ b/perf_test/sparse/KokkosSparse_gs.cpp
@@ -57,6 +57,22 @@
 using std::cout;
 using std::string;
 
+#if defined(KOKKOSKERNELS_INST_ORDINAL_INT)
+  typedef int default_lno_t;
+#elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
+  typedef int64_t default_lno_t;
+#else
+  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+#endif
+  //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
+#if defined(KOKKOSKERNELS_INST_OFFSET_INT)
+  typedef int default_size_type;
+#elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
+  typedef size_t default_size_type;
+#else
+  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
+#endif
+
 template<typename size_type, typename lno_t, typename device_t>
 void runGS(string matrixPath, string devName, bool symmetric)
 {
@@ -223,28 +239,28 @@ int main(int argc, char** argv)
   #ifdef KOKKOS_ENABLE_SERIAL
   if(device == "serial")
   {
-    runGS<size_t, int, Kokkos::Serial>(matrixPath, device, sym);
+    runGS<default_size_type, default_lno_t, Kokkos::Serial>(matrixPath, device, sym);
     run = true;
   }
   #endif
   #ifdef KOKKOS_ENABLE_OPENMP
   if(device == "openmp")
   {
-    runGS<size_t, int, Kokkos::OpenMP>(matrixPath, device, sym);
+    runGS<default_size_type, default_lno_t, Kokkos::OpenMP>(matrixPath, device, sym);
     run = true;
   }
   #endif
   #ifdef KOKKOS_ENABLE_THREADS
   if(device == "threads")
   {
-    runGS<size_t, int, Kokkos::Threads>(matrixPath, device, sym);
+    runGS<default_size_type, default_lno_t, Kokkos::Threads>(matrixPath, device, sym);
     run = true;
   }
   #endif
   #ifdef KOKKOS_ENABLE_CUDA
   if(device == "cuda")
   {
-    runGS<size_t, int, Kokkos::Cuda>(matrixPath, device, sym);
+    runGS<default_size_type, default_lno_t, Kokkos::Cuda>(matrixPath, device, sym);
     run = true;
   }
   #endif

--- a/perf_test/sparse/KokkosSparse_gs.cpp
+++ b/perf_test/sparse/KokkosSparse_gs.cpp
@@ -62,7 +62,7 @@ using std::string;
 #elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
   typedef int64_t default_lno_t;
 #else
-  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+  #error "Expect int and/or int64_t to be enabled as ORDINAL (lno_t) types"
 #endif
   //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
 #if defined(KOKKOSKERNELS_INST_OFFSET_INT)
@@ -70,7 +70,7 @@ using std::string;
 #elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
   typedef size_t default_size_type;
 #else
-  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
+  #error "Expect size_t and/or int to be enabled as OFFSET (size_type) types"
 #endif
 
 template<typename size_type, typename lno_t, typename device_t>

--- a/perf_test/sparse/KokkosSparse_spiluk.cpp
+++ b/perf_test/sparse/KokkosSparse_spiluk.cpp
@@ -70,7 +70,7 @@
 #elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
   typedef int64_t default_lno_t;
 #else
-  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+  #error "Expect int and/or int64_t to be enabled as ORDINAL (lno_t) types"
 #endif
   //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
 #if defined(KOKKOSKERNELS_INST_OFFSET_INT)
@@ -78,7 +78,7 @@
 #elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
   typedef size_t default_size_type;
 #else
-  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
+  #error "Expect size_t and/or int to be enabled as OFFSET (size_type) types"
 #endif
 
 #if defined( KOKKOS_ENABLE_CXX11_DISPATCH_LAMBDA ) && (!defined(KOKKOS_ENABLE_CUDA) || ( 8000 <= CUDA_VERSION ))

--- a/perf_test/sparse/KokkosSparse_spiluk.cpp
+++ b/perf_test/sparse/KokkosSparse_spiluk.cpp
@@ -65,6 +65,22 @@
 #include "KokkosSparse_CrsMatrix.hpp"
 #include <KokkosKernels_IOUtils.hpp>
 
+#if defined(KOKKOSKERNELS_INST_ORDINAL_INT)
+  typedef int default_lno_t;
+#elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
+  typedef int64_t default_lno_t;
+#else
+  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+#endif
+  //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
+#if defined(KOKKOSKERNELS_INST_OFFSET_INT)
+  typedef int default_size_type;
+#elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
+  typedef size_t default_size_type;
+#else
+  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
+#endif
+
 #if defined( KOKKOS_ENABLE_CXX11_DISPATCH_LAMBDA ) && (!defined(KOKKOS_ENABLE_CUDA) || ( 8000 <= CUDA_VERSION ))
 using namespace KokkosSparse;
 using namespace KokkosSparse::Experimental;
@@ -77,8 +93,8 @@ template<typename Scalar>
 int test_spiluk_perf(std::vector<int> tests, std::string afilename, int k, int team_size, int vector_length, /*int idx_offset,*/ int loop) {
 
   typedef Scalar scalar_t;
-  typedef int lno_t;
-  typedef int size_type;
+  typedef default_lno_t lno_t;
+  typedef default_size_type size_type;
   typedef Kokkos::DefaultExecutionSpace execution_space;
   typedef typename execution_space::memory_space memory_space;
 
@@ -111,6 +127,8 @@ int test_spiluk_perf(std::vector<int> tests, std::string afilename, int k, int t
     const typename KernelHandle::const_nnz_lno_t fill_lev = lno_t(k) ;
 
 #ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+    //cuSPARSE requires lno_t = size_type = int. For both, int is always used (if enabled)
+#if defined(KOKKOSKERNELS_INST_ORDINAL_INT) && defined(KOKKOSKERNELS_INST_OFFSET_INT)
     //std::cout << "  cusparse: create handle" << std::endl;
     cusparseStatus_t status;
     cusparseHandle_t handle = 0;
@@ -143,6 +161,9 @@ int test_spiluk_perf(std::vector<int> tests, std::string afilename, int k, int t
               descr, A.values.data(), A.graph.row_map.data(), A.graph.entries.data(), info, &pBufferSize);
     // pBuffer returned by cudaMalloc is automatically aligned to 128 bytes.
     cudaMalloc((void**)&pBuffer, pBufferSize);
+#else
+    std::cout << "Note: the cuSPARSE TPL is enabled, but either offset=int or ordinal=int is disabled, so it can't be used.\n";
+#endif
 #endif
 
     for ( auto test : tests ) {
@@ -496,6 +517,7 @@ int main(int argc, char **argv)
 }
 #else
 int main() {
+  std::cout << "The SPILUK perf_test requires CUDA >= 8.0\n";
   return 0;
 }
 #endif

--- a/perf_test/sparse/KokkosSparse_spmv_struct.cpp
+++ b/perf_test/sparse/KokkosSparse_spmv_struct.cpp
@@ -65,7 +65,7 @@ enum {AUTO, DYNAMIC, STATIC};
 #elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
   typedef int64_t default_lno_t;
 #else
-  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+  #error "Expect int and/or int64_t to be enabled as ORDINAL (lno_t) types"
 #endif
   //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
 #if defined(KOKKOSKERNELS_INST_OFFSET_INT)
@@ -73,7 +73,7 @@ enum {AUTO, DYNAMIC, STATIC};
 #elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
   typedef size_t default_size_type;
 #else
-  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
+  #error "Expect size_t and/or int to be enabled as OFFSET (size_type) types"
 #endif
 
 void print_help() {

--- a/perf_test/sparse/KokkosSparse_spmv_struct.cpp
+++ b/perf_test/sparse/KokkosSparse_spmv_struct.cpp
@@ -60,10 +60,20 @@
 enum {STRUCT, UNSTR};
 enum {AUTO, DYNAMIC, STATIC};
 
-#ifdef INT64
-typedef long long int LocalOrdinalType;
+#if defined(KOKKOSKERNELS_INST_ORDINAL_INT)
+  typedef int default_lno_t;
+#elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
+  typedef int64_t default_lno_t;
 #else
-typedef int LocalOrdinalType;
+  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+#endif
+  //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
+#if defined(KOKKOSKERNELS_INST_OFFSET_INT)
+  typedef int default_size_type;
+#elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
+  typedef size_t default_size_type;
+#else
+  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
 #endif
 
 void print_help() {
@@ -140,18 +150,18 @@ int main(int argc, char **argv)
   }
 
   Kokkos::initialize(argc,argv);
-
   {
-
-    typedef KokkosSparse::CrsMatrix<Scalar,int,Kokkos::DefaultExecutionSpace,void,int> matrix_type;
+    typedef default_lno_t lno_t;
+    typedef default_size_type size_type;
+    typedef KokkosSparse::CrsMatrix<Scalar,lno_t,Kokkos::DefaultExecutionSpace,void,size_type> matrix_type;
     typedef typename Kokkos::View<Scalar**,Kokkos::LayoutLeft> mv_type;
     // typedef typename Kokkos::View<Scalar*,Kokkos::LayoutLeft,Kokkos::MemoryRandomAccess > mv_random_read_type;
     typedef typename mv_type::HostMirror h_mv_type;
 
     int leftBC = 1, rightBC = 1, frontBC = 1, backBC = 1, bottomBC = 1, topBC = 1;
 
-    Kokkos::View<int*, Kokkos::HostSpace> structure("Spmv Structure", numDimensions);
-    Kokkos::View<int*[3], Kokkos::HostSpace> mat_structure("Matrix Structure", numDimensions);
+    Kokkos::View<lno_t*, Kokkos::HostSpace> structure("Spmv Structure", numDimensions);
+    Kokkos::View<lno_t*[3], Kokkos::HostSpace> mat_structure("Matrix Structure", numDimensions);
     if(numDimensions == 1) {
       structure(0) = nx;
       mat_structure(0, 0) = nx;
@@ -264,8 +274,8 @@ int main(int argc, char **argv)
 
       double problem_size = matrix_size+vector_size;
       printf("Type NNZ NumRows NumCols ProblemSize(MB) AveBandwidth(GB/s) MinBandwidth(GB/s) MaxBandwidth(GB/s) AveGFlop MinGFlop MaxGFlop aveTime(ms) maxTime(ms) minTime(ms)\n");
-      printf("Struct %i %i %i %6.2lf ( %6.2lf %6.2lf %6.2lf ) ( %6.3lf %6.3lf %6.3lf ) ( %6.3lf %6.3lf %6.3lf )\n",
-	     A.nnz(), A.numRows(), A.numCols(), problem_size,
+      printf("Struct %zu %zu %zu %6.2lf ( %6.2lf %6.2lf %6.2lf ) ( %6.3lf %6.3lf %6.3lf ) ( %6.3lf %6.3lf %6.3lf )\n",
+	     (size_t) A.nnz(), (size_t) A.numRows(), (size_t) A.numCols(), problem_size,
 	     (matrix_size+vector_readwrite)/ave_time*loop/1024, (matrix_size+vector_readwrite)/max_time/1024, (matrix_size+vector_readwrite)/min_time/1024,
 	     2.0*A.nnz()*loop/ave_time/1e9, 2.0*A.nnz()/max_time/1e9, 2.0*A.nnz()/min_time/1e9,
 	     ave_time/loop*1000, max_time*1000, min_time*1000);
@@ -294,8 +304,8 @@ int main(int argc, char **argv)
       double vector_readwrite = (A.nnz() + A.numCols())*sizeof(Scalar)/1024/1024;
 
       double problem_size = matrix_size+vector_size;
-      printf("Unstr %i %i %i %6.2lf ( %6.2lf %6.2lf %6.2lf ) ( %6.3lf %6.3lf %6.3lf ) ( %6.3lf %6.3lf %6.3lf )\n",
-	     A.nnz(), A.numRows(),A.numCols(), problem_size,
+      printf("Unstr %zu %zu %zu %6.2lf ( %6.2lf %6.2lf %6.2lf ) ( %6.3lf %6.3lf %6.3lf ) ( %6.3lf %6.3lf %6.3lf )\n",
+	     (size_t) A.nnz(), (size_t) A.numRows(), (size_t) A.numCols(), problem_size,
 	     (matrix_size+vector_readwrite)/ave_time*loop/1024, (matrix_size+vector_readwrite)/max_time/1024,(matrix_size+vector_readwrite)/min_time/1024,
 	     2.0*A.nnz()*loop/ave_time/1e9, 2.0*A.nnz()/max_time/1e9, 2.0*A.nnz()/min_time/1e9,
 	     ave_time/loop*1000, max_time*1000, min_time*1000);

--- a/perf_test/sparse/KokkosSparse_sptrsv.cpp
+++ b/perf_test/sparse/KokkosSparse_sptrsv.cpp
@@ -69,7 +69,7 @@
 #elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
   typedef int64_t default_lno_t;
 #else
-  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+  #error "Expect int and/or int64_t to be enabled as ORDINAL (lno_t) types"
 #endif
   //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
 #if defined(KOKKOSKERNELS_INST_OFFSET_INT)
@@ -77,7 +77,7 @@
 #elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
   typedef size_t default_size_type;
 #else
-  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
+  #error "Expect size_t and/or int to be enabled as OFFSET (size_type) types"
 #endif
 
 #if defined( KOKKOS_ENABLE_CXX11_DISPATCH_LAMBDA ) && (!defined(KOKKOS_ENABLE_CUDA) || ( 8000 <= CUDA_VERSION ))

--- a/perf_test/sparse/KokkosSparse_sptrsv.cpp
+++ b/perf_test/sparse/KokkosSparse_sptrsv.cpp
@@ -64,6 +64,22 @@
 #include "KokkosSparse_CrsMatrix.hpp"
 #include <KokkosKernels_IOUtils.hpp>
 
+#if defined(KOKKOSKERNELS_INST_ORDINAL_INT)
+  typedef int default_lno_t;
+#elif defined(KOKKOSKERNELS_INST_ORDINAL_INT64_T)
+  typedef int64_t default_lno_t;
+#else
+  #error "Expect INT and/or INT64_T to be enabled as ORDINAL (lno_t) types"
+#endif
+  //Prefer int as the default offset type, because cuSPARSE doesn't support size_t for rowptrs.
+#if defined(KOKKOSKERNELS_INST_OFFSET_INT)
+  typedef int default_size_type;
+#elif defined(KOKKOSKERNELS_INST_OFFSET_SIZE_T)
+  typedef size_t default_size_type;
+#else
+  #error "Expect SIZE_T and/or INT to be enabled as OFFSET (size_type) types"
+#endif
+
 #if defined( KOKKOS_ENABLE_CXX11_DISPATCH_LAMBDA ) && (!defined(KOKKOS_ENABLE_CUDA) || ( 8000 <= CUDA_VERSION ))
 using namespace KokkosSparse;
 using namespace KokkosSparse::Experimental;
@@ -78,8 +94,8 @@ template<typename Scalar>
 int test_sptrsv_perf(std::vector<int> tests, std::string& lfilename, std::string& ufilename, int team_size, int vector_length, int idx_offset, int loop) {
 
   typedef Scalar scalar_t;
-  typedef int lno_t;
-  typedef int size_type;
+  typedef default_lno_t lno_t;
+  typedef default_size_type size_type;
   typedef Kokkos::DefaultExecutionSpace execution_space;
   typedef typename execution_space::memory_space memory_space;
 

